astropy/utils/masked/function_helpers.py

```diff
@@ -133,10 +133,17 @@
 }  # fmt: skip
 
 # Really should do these...
-IGNORED_FUNCTIONS |= {
-    getattr(np, setopsname) for setopsname in np.lib.arraysetops.__all__
-}
-
+if NUMPY_LT_2_0:
+    NP_LIB_ARRAYSETOP_FUNCS = np.lib.arraysetops.__all__
+else:
+    # np.lib.arraysetops is private in numpy 2.0 and raises AttributeError
+    # substitute a hard-coded copy for now
```

Comment:
```suggestion
    # public set operations from have been moved to the top-level namespace in numpy 2.0
    # (numpy/numpy#24507), raising an AttributeError when accessed through np.lib.arraysetops.
    # Listing them explicitly here.
```
Thanks for the fix; just some clarification to the comment (as to why they still need to be excluded at all).
Note that `numpy.lib.arraysetops` could still be imported explicitly, but from https://github.com/numpy/numpy/pull/24567#discussion_r1310266610 this looks like the right solution.